### PR TITLE
SC-6293 - adds learn store permission

### PR DESCRIPTION
**Pull request by virgilchiriac**
# Description

There was the need to dissallow students from Schleswig-Holstein access to the Lern-Store. Since there is no way to grant access to some students, but not others, the proposed solution is using an additional role and extends the permission system to allow for negative permission checks. I.e. access is granted if users have LERNSTORE_VIEW permission and don't have LERNSTORE_DENY permission.

## Links to Tickets or other pull requests

- https://ticketsystem.hpi-schul-cloud.org/browse/SC-6293
- https://github.com/hpi-schul-cloud/schulcloud-client/pull/2170
- https://github.com/hpi-schul-cloud/schulcloud-server/pull/1718

## Changes

Permissions can be checked for combinations of conditions and now supports NOT operator. 
For details check docs/1-Tutorials/PagePermissionCheck.md

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ Allowed Types of change: `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `
 
 ## Unreleased
 
+- SC-6239 - adds lernstore permmission
```
Comment (JenyMzo) [inline CHANGELOG.md:11]: permission*

File: src/middleware/permission-check.js [hunk 2]
```diff
@@ -11,16 +11,38 @@ export default async ({ app, store, route }) => {
...
-		const OPERATOR = Array.isArray(meta.requiredPermissions)
+		const operator = Array.isArray(meta.requiredPermissions)
 			? "AND"
-			: meta.requiredPermissions.operator;
-		return OPERATOR === "AND"
-			? REQUIRED_PERMISSIONS.every(userHasPermission)
-			: REQUIRED_PERMISSIONS.some(userHasPermission);
+			: meta.requiredPermissions.operator.toUpperCase();
+
+		const OPERATORS = ["OR", "AND", "XOR", "NOT", "!"];
```
Comment (JenyMzo) [inline src/middleware/permission-check.js:22]: Why are you changing the constants to lowercase here but operators is uppercase? should we keep a convention?

File: docs/1-Tutorials/PagePermissionCheck.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ export default {
 
 In this case, the current user must have all the defined permissions.
 
-If you wan't more controle you can pass an Object:
+b) If you wan't more controle you can pass an Object:
```
Comment (JenyMzo) [inline docs/1-Tutorials/PagePermissionCheck.md:19]: control?

**Review by JenyMzo — APPROVED** [review]

**Comment by virgilchiriac** [comment]
branch name was wrong, so I created another one and closing this one
https://github.com/hpi-schul-cloud/nuxt-client/pull/1375

**Pull request closed without merging**